packages/botonic-plugin-flow-builder/src/api.ts

```diff
@@ -176,4 +179,27 @@ export class FlowBuilderApi {
   private containsAnyKeywords(input: string, keywords: string[]): boolean {
     return keywords.some(keyword => input.includes(keyword))
   }
+
+  getPayloadFromBotActionNodeId(id?: string): string | undefined {
+    if (id) {
+      const botActionNode = this.getNodeById(id)
+      if (botActionNode.type === HtNodeWithoutContentType.BOT_ACTION) {
+        return this.createPayloadWithParams(botActionNode)
+      }
+    }
+    return undefined
+  }
```

Comment:
Why are you defining the `id` param as optional and then returning undefined if it is not defined?

Reply:
I have refactored this as we have discussed with Adria it will now be a function called getPayload and will return the payload. It will create the payload from a BotAction node or return the id of the text, image, video, carousel etc that is connected to a button, handoff, intent or keyword.

Reply:
Yes I was confused, sorry for the noise